query_panel/main.js

```diff
@@ -34,6 +34,75 @@ const app = createApp({
     };
   },
   methods: {
+    // Converts the provided data to CSV format.
+    dataToCsv(data) {
+      try {
+        if (!data || data.length === 0) {
+          console.error("No data available to convert to CSV");
+          return "";
+        }
+        const replacer = (key, value) => (value === null ? "" : value);
+        const header = Object.keys(data[0]);
+        const csv = [
+          header.join(","),
+          ...data.map((row) =>
+            header
+              .map((fieldName) => {
+                let fieldData = row[fieldName];
+                if (fieldData && typeof fieldData === "string") {
+                  fieldData = fieldData.replace(/"/g, '""'); // Escape double quotes
+                  return `"${fieldData}"`; // Wrap in double quotes
+                }
+                return JSON.stringify(fieldData, replacer);
+              })
+              .join(","),
+          ),
+        ].join("\r\n");
+        return csv;
+      } catch (error) {
+        console.error("Error converting data to CSV:", error);
+        return ""; // Return an empty string if there's an error
+      }
+    },
+    downloadAsCSV() {
+      const data = this.table.getData(); // Get the data the same way you do for copying
+      if (!data || data.length === 0) {
+        console.error("No data available for downloading.");
+        return;
+      }
+      let csvContent = this.dataToCsv(data);
+      let blob = new Blob([csvContent], { type: "text/csv" });
+      let url = window.URL.createObjectURL(blob);
+      let a = document.createElement("a");
+      a.href = url;
+      a.download = `power_user_data_${new Date().toISOString()}.csv`; // Filename with a timestamp
+      a.click();
+    },
+    copyTextToClipboard(text) {
+      const el = document.createElement("textarea");
```

Comment:
Why use dynamic html and not use s standard vscode-textarea and vue data element.